src/app/settings/keyboard-mapping-settings/components/keyboard-mapping-settings-page/keyboard-mapping-settings-page.component.ts

```diff
@@ -0,0 +1,222 @@
+import { Logger } from 'src/app/core/abstractions/logger.service'
+import { Component, OnDestroy, OnInit } from '@angular/core'
+import { ActionTrigger } from 'src/app/shared/models/action-trigger'
+import { ActionTriggerStateService } from 'src/app/core/services/action-trigger-state.service'
+import { Subject, takeUntil } from 'rxjs'
+import { Tv2PartAction } from 'src/app/shared/models/tv2-action'
+import { ActionStateService } from 'src/app/shared/services/action-state.service'
+import { KeyboardTriggerData } from 'src/app/shared/models/keyboard-trigger-data'
+import { SofieTableHeader, SortDirection } from '../../../../shared/components/table/table.component'
+import { IconButton, IconButtonSize } from '../../../../shared/enums/icon-button'
+import { DialogService } from '../../../../shared/services/dialog.service'
+import { ActionTriggerService } from '../../../../shared/abstractions/action-trigger.service'
+import { NotificationService } from '../../../../shared/services/notification.service'
+import { ActionTriggerParser } from '../../../../shared/abstractions/action-trigger-parser.service'
+import { FormatKeyboardKeysPipe } from '../../../../shared/pipes/format-keyboard-keys.pipe'
+import { EditKeyboardMappingDialogComponent } from '../edit-keyboard-mapping-dialog/edit-keyboard-mapping-dialog.component'
+
+export interface KeyboardMapping {
+  actionTrigger: ActionTrigger<KeyboardTriggerData>
+  action: Tv2PartAction
+}
+
+@Component({
+  selector: 'sofie-keyboard-mapping-settings-page',
+  templateUrl: './keyboard-mapping-settings-page.component.html',
+})
+export class KeyboardMappingSettingsPageComponent implements OnInit, OnDestroy {
+  protected readonly IconButtonSize = IconButtonSize
+  protected readonly IconButton = IconButton
+  protected readonly title: string = $localize`settings.keyboard-mappings.label`
+  protected readonly keyboardMappingsFileName: string = 'keyboard-mappings'
+
+  public keyboardMappings: KeyboardMapping[]
+
+  public selectedKeyboardMappings: Set<KeyboardMapping> = new Set()
+
+  public readonly headers: SofieTableHeader<KeyboardMapping>[] = [
+    {
+      name: 'Label',
+      isBeingUsedForSorting: true,
+      sortCallback: (a: KeyboardMapping, b: KeyboardMapping): number => a.actionTrigger.data.label.localeCompare(b.actionTrigger.data.label),
+      sortDirection: SortDirection.DESC,
+    },
+    {
+      name: 'Shortcut',
+      sortCallback: (a: KeyboardMapping, b: KeyboardMapping): number => a.actionTrigger.data.mappedToKeys?.toString().localeCompare(b.actionTrigger.data.mappedToKeys?.toString() ?? '') ?? 0,
+      sortDirection: SortDirection.DESC,
+    },
+    {
+      name: 'Action',
+      sortCallback: (a: KeyboardMapping, b: KeyboardMapping): number => a.action.name.localeCompare(b.action.name),
+      sortDirection: SortDirection.DESC,
+    },
+  ]
```

Comment:
This sortCallback and filter functionality hide a risk of calling angular detection too often because we're sending functions to the template that it rerenders every time on changes. I need a little more time to look at the logic and determine if there is a potential problem just mentioning it as something that needs to be discussed.

Reply:
It's been noted.
But as we talked about, the `TableComponent` could make a more sophisticated filtering where they still used this sort callback, so that calling components of the TableComponent wouldn't need to worry about filter logic.